src/content/docs/codestream/how-use-codestream/performance-monitoring.mdx

```diff
@@ -19,6 +19,66 @@ Go here to get or create your [New Relic user key](https://one.newrelic.com/laun
 
 Once you have your user key, in **Observability** click **Connect to New Relic One**, then paste your user key and click **Connect**.
 
+## Monitor code-level metrics [#code-level]
+
+CodeStream's code-level metrics give you detailed insight into how your code is performing.
+
+![A screenshot of a Python method with average duration, throughput, and error rate telemetry.](./images/code-level-metrics.png "A Python method with average duration, throughput, and error rate metrics.")
+
+<figcaption>When relevant, CodeStream's code-level metrics are shown in a CodeLens above your Python methods.</figcaption>
+
+<Callout variant="caution">
+Code-level metrics are only available for Python code and [supported Python frameworks](/docs/apm/agents/python-agent/getting-started/instrumented-python-packages). These metrics are shown via CodeLens text, a line of text inserted above your instrumented method.
+
+Your Python application code also needs to be monitored by the Python APM agent and your repository associated with a New Relic service.
+</Callout>
+
+Click the CodeLens text to see related charts.
+
+![A screenshot of an expanded code-level metrics CodeLens that shows related charts.](./images/code-level-charts.png "An expanded code-level metrics CodeLens that shows related charts.")
+
+<figcaption>Click the entity name at the top of the charts to jump to this data in New Relic One.</figcaption>
+
+Code-level metrics focus on your "golden metrics" for the last 30 minutes. These metrics are: latency, traffic, errors, and saturation.
```

Comment:
I seem to be very pro-links today. Maybe we should link out to a doc that defines `golden metrics` as a CodeStream user new to observability may not know what that means.

Reply:
Scratch that, I should have read the second sentence 😅